posthog/temporal/batch_exports/postgres_batch_export.py

```diff
@@ -254,6 +254,27 @@ async def adelete_table(self, schema: str | None, table_name: str, not_found_ok:
 
                 await cursor.execute(sql.SQL(base_query).format(table=table_identifier))
 
+    async def aget_table_columns(self, schema: str | None, table_name: str) -> list[str]:
+        """Get the column names for a table in PostgreSQL.
+
+        Args:
+            schema: Name of the schema where the table is located.
+            table_name: Name of the table to get columns for.
+
+        Returns:
+            A list of column names in the table.
+        """
+        if schema:
+            table_identifier = sql.Identifier(schema, table_name)
+        else:
+            table_identifier = sql.Identifier(table_name)
+
+        async with self.connection.transaction():
+            async with self.connection.cursor() as cursor:
+                await cursor.execute(sql.SQL("SELECT * FROM {} WHERE 1=0").format(table_identifier))
```

Comment:
That's a very funny way to get schema information.

Reply:
That was what my editor suggested 😄 
I suppose it is efficient since it won't actually return any table data, just column metadata

Reply:
It's a bit obfuscated, maybe not so clear at a glace what you are doing, but I believe it does work. And it saves you from needing to query a separate table (something in `information_schema`) for which we may need extra permissions.